### ISSUE TITLE
UpperBound Experiments ERROR

### ISSUE THREAD

**Issue by yhl2018** [Jwoo5/ecg-qa#4]
Hey, thanks for the code and the QA dataset.

I am trying to reproduce the results in the paper for upperbound experiments, using the PTB-XL dataset, following every step as mentioned in the README, but get error report：

num_labels=83

fairseq-hydra-train task.data=datasets/preprocess_upperbound \
    model.num_labels=$num_labels \
    --config-dir examples/scratch/ecg_classification/resnet \
    --config-name se_wrn_total

![Image](https://github.com/user-attachments/assets/7468d416-ae8e-4968-9de6-1ab0f93b9a23)

**Comment by yhl2018** [comment]
Pre-process for Upperbound experiments  

![Image](https://github.com/user-attachments/assets/bfe777dd-4b77-426e-95f8-5c9daa9a95b1)  

![Image](https://github.com/user-attachments/assets/f0bc760c-2a36-4431-b04a-2f674b7b150a)

**Comment by Jwoo5** [comment]
Thanks for reporting this issue.
Seems that it is due to the recent updates on the structure of the criterions in the fairseq-signals repo.
I will fix it and let you know soon.

**Comment by yhl2018** [comment]
The problem still exists even after the fix.  ![Image](https://github.com/user-attachments/assets/22e21d84-2ee3-4a30-b1b2-1d427b03936d)

**Comment by yhl2018** [comment]
> The problem still exists even after the fix. ![Image](https://github.com/user-attachments/assets/22e21d84-2ee3-4a30-b1b2-1d427b03936d)

@Jwoo5

**Comment by Jwoo5** [comment]
Seems that `torch.tensor(target).squeeze().float().cuda()` instead of `torch.cat(target).float()` will fix this as a temporary hack, but I am finding a more general solution for this.

**Comment by Jwoo5** [comment]
I found that there was a minor error when loading the label, and fixed it accordingly.
Please try again in `dev` branch of the `fairseq-signals`.
If it works well now, I will merge it to the master branch and close this issue.
Tagging @yhl2018 and @asdfwqfu to track this issue.

**Comment by asdfwqfu** [comment]
> I found that there was a minor error when loading the label, and fixed it accordingly. Please try again in branch of the . If it works well now, I will merge it to the master branch and close this issue. Tagging [@yhl2018](https://github.com/yhl2018) and [@asdfwqfu](https://github.com/asdfwqfu) to track this issue.`dev``fairseq-signals`

Professor, I wanted to let you know that it's now running successfully. Thank you for your help.

**Comment by Jwoo5** [comment]
Thanks for confirming it!
